server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -145,8 +94,9 @@
   // Record the username for every rpc connection. Username.get() is null if
   // login is failed.
   protected ThreadLocal<String> username = new ThreadLocal<>();
-  private ThreadLocal<HashMap<String, PhysicalPlan>> queryStatus = new ThreadLocal<>();
-  private ThreadLocal<HashMap<String, QueryDataSet>> queryRet = new ThreadLocal<>();
+  protected ThreadLocal<AtomicLong> queryId = new ThreadLocal<>();
+  private ThreadLocal<HashMap<Long, PhysicalPlan>> queryStatus = new ThreadLocal<>();
```

Comment:
Maybe a structure QueryContext could be used, which contains a query id, PhysicalPlan and QueryDataSet. If you put these fields here, it's better to add some comments. For example, queryStatus means a queryId -> PhysicalPlan.

Reply:
QueryContext only contains the query operation's context. Actually, The queryStatus should also contains other operations' info like insert, update. Maybe, I should change the name of queryStatus to operationStatus.
And the QueryContext will be created until the phase fetchResults, but queryStatus will be created when executeStatement.